dockerfiles/Dockerfile.build

```diff
@@ -1,17 +1,20 @@
-FROM ubuntu:focal-20220531 AS build
+ARG base_default=ghcr.io/edgelesssys/ego/build-base:v0.5.1
+ARG base=$base_default
+FROM $base AS build
+ARG base_default
+ARG base
 
-RUN apt update && DEBIAN_FRONTEND=noninteractive apt install -y \
-  build-essential=12.8ubuntu1.1 \
-  clang-10=1:10.0.0-4ubuntu1 \
-  cmake=3.16.3-1ubuntu1 \
+RUN test "$base" = "$base_default" || apt-get update \
```

Comment:
Would there be any argument against completely throwing this out? The multiple arguments and conditional executions makes the Dockerfile a bit complicated to read. I think if someone really wants to build with Ubuntu, they can just paste these two files together. Or we could even put both stages into one file, I think that would still be better to read rather than multiple arguments.

Reply:
> The multiple arguments and conditional executions make the Dockerfile a bit complicated to read.

Agreed. The least I should do is adding comments.

> if someone really wants to build with Ubuntu, they can just paste these two files together.

I thought there should be a way to build this without any "precompiled" Edgeless artifacts with a single command, but maybe a bit of editing might be acceptable. I'm still undecided yet.

> Or we could even put both stages into one file

How would this work to have the stage in this file, but still pulling it by default?